paddle/fluid/framework/tensor.h

```diff
@@ -45,10 +45,11 @@ class Tensor {
   friend struct EigenVector;
 
  public:
-  Tensor() : offset_(0) {}
+  Tensor() : offset_(0), use_pinned_(false) {}
```

Comment:
Curious why need to add `use_pinned_ ` to tensor, if we just want to use pinned memory for copying, just put this in the `CopyTensor` implement should OK?

Reply:
No of course. I was thinking about GPU->CPU copy. For CPU->GPU case, it seems this is the only way now.

Reply:
Why do we use the pinned memory just as a staging area but not use it directly? The computations of CPU can access the data of the pinned memory directly.

Reply:
In that case, can we set `use_pinned_` as a global GFLAG, so that all allocations on host would use pinned memory, then we can test the overall performance boost.

Reply:
Because the memory copying between GPU and CPU is async, for GPU->CPU case, before reading data from the pinned memory we should ensure the copy has completed, so we should add sync operation. So using pinned memory is a little complex. 
I plan to only put the input data into pinned memory and test the overall performance.

Reply:
Is this still WIP, can I review and merge this?